packages/grid/x-data-grid/src/components/toolbar/GridToolbarQuickFilter.tsx

```diff
@@ -39,12 +65,39 @@ export type GridToolbarQuickFilterProps = TextFieldProps & {
 };
 
 function GridToolbarQuickFilter(props: GridToolbarQuickFilterProps) {
-  const { quickFilterParser = defaultSearchValueParser, debounceMs = 500, ...other } = props;
+  const {
+    quickFilterParser = defaultSearchValueParser,
+    quickFilterFormatter = defaultSearchValueFormatter,
+    debounceMs = 500,
+    ...other
+  } = props;
 
   const apiRef = useGridApiContext();
   const rootProps = useGridRootProps();
+  const quickFilterValues = gridQuickFilterValuesSelector(apiRef);
 
-  const [searchValue, setSearchValue] = React.useState('');
+  const [searchValue, setSearchValue] = React.useState(
+    quickFilterFormatter(quickFilterValues ?? []),
```

Comment:
To not call `quickFilterFormatter` on every render.

```suggestion
    () => quickFilterFormatter(quickFilterValues ?? []),
```

Reply:
I don't get why. The default value provided in `React.useState` should be computed only in the first render. After the first render, this parameter has no effect on the value of the state